src/libfuncs/cast.rs

```diff
@@ -421,74 +319,51 @@ pub fn build_upcast<'ctx, 'this>(
+                ))?;
+
                 is_not_neg_block.append_operation(cf::br(final_block, &[result], location));
             }
 
             {
-                let mut result = is_neg_block
-                    .append_operation(arith::extsi(
-                        entry.argument(0)?.into(),
-                        IntegerType::new(context, dst_width.try_into()?).into(),
-                        location,
-                    ))
-                    .result(0)?
-                    .into();
-                let prime = is_neg_block
-                    .append_operation(arith::constant(
-                        context,
-                        Attribute::parse(
-                            context,
-                            &format!(
-                                "{} : {}",
-                                metadata
-                                    .get::<PrimeModuloMeta<Felt>>()
-                                    .ok_or(Error::MissingMetadata)?
-                                    .prime(),
-                                dst_type
-                            ),
-                        )
-                        .ok_or(Error::ParseAttributeError)?,
-                        location,
-                    ))
-                    .result(0)?
-                    .into();
-                result = is_neg_block
-                    .append_operation(arith::addi(result, prime, location))
-                    .result(0)?
-                    .into();
+                let mut result = is_neg_block.append_op_result(arith::extsi(
+                    entry.argument(0)?.into(),
+                    IntegerType::new(context, dst_width.try_into()?).into(),
+                    location,
+                ))?;
+                let prime = is_neg_block.const_int_from_type(
+                    context,
+                    location,
+                    metadata
+                        .get::<PrimeModuloMeta<Felt>>()
+                        .ok_or(Error::MissingMetadata)?
+                        .prime(),
+                    dst_type,
+                )?;
+
+                result = is_neg_block.append_op_result(arith::addi(result, prime, location))?;
                 is_neg_block.append_operation(cf::br(final_block, &[result], location));
             }
 
             let result = final_block.argument(0)?.into();
             final_block.append_operation(helper.br(0, &[result], location));
             return Ok(());
         } else {
-            block
-                .append_operation(arith::extsi(
-                    entry.argument(0)?.into(),
-                    IntegerType::new(context, dst_width.try_into()?).into(),
-                    location,
-                ))
-                .result(0)?
-                .into()
-        }
-    } else {
-        block
-            .append_operation(arith::extui(
-                block.argument(0)?.into(),
+            block.append_operation(arith::extsi(
```

Comment:
here it should be append_op_result, because the result is used, when returned from this else branch